src/Nethermind/Nethermind.Synchronization/SnapSync/SnapProvider.cs

```diff
@@ -261,6 +263,18 @@ public void RetryRequest(SnapSyncBatch batch)
 
         }
 
-        public bool IsSnapGetRangesFinished() => _progressTracker.IsSnapGetRangesFinished();
+        private bool _isSnapGetRangesFinished;
+        public bool IsSnapGetRangesFinished() => _isSnapGetRangesFinished || CheckIfSnapGetRangesFinished();
+        private bool CheckIfSnapGetRangesFinished()
+        {
+            _isSnapGetRangesFinished = _progressTracker.IsSnapGetRangesFinished();
+
+            if (_isSnapGetRangesFinished)
+            {
+                RemoveSnapCapability?.Invoke(this, EventArgs.Empty);
+            }
+
+            return _isSnapGetRangesFinished;
+        }
```

Comment:
What if we need to restart state sync? Wouldn't it be better to base it on MultiSyncModeSelector changing state to State sync?